profile/profile.go

```diff
@@ -198,7 +198,7 @@ func (p *Profile) ValidOwnerProfile() error {
 // to the profile ID if none is present
 func (p *Profile) GetKeyID() key.ID {
 	if p.KeyID == key.ID("") {
-		p.KeyID = key.ID(p.ID.String())
+		p.KeyID = key.ID(string(p.ID))
```

Comment:
Am I wrong or is this a change in functionality? Looking at https://github.com/libp2p/go-libp2p-core/blob/master/peer/peer.go#L56, `peer.ID.String()` calls `Pretty()` which calls `IDB58Encode`. However, I'm pretty sure `string(p.ID)` just coerces the bytes of the id to be a string without any additional encoding. Would love to have a unit test specifically on this function to prove me wrong/right.

Reply:
Yes I totally agree a proper set of tests are needed. This is actually fixing a bug introduced earlier, as key.ID has become an alias for peer.ID. I plan to fix _all_ of this & add tests in a follow-up

Reply:
I should add that this PR does help somewhat on this frontier by requiring that keys added adding a key to the keystore now **requires** a match between the ID & public key, so if the id is malformed, it'll fail.

Reply:
Ah, you're saying this *was* using encode before, but that was a bug, and it should only be coercing the bytes? Ok, that makes sense. Unfortunate how it's hard to tell at the moment.